Remove view bounds

OlivierBlanvillain opened the pull request:
i.e. use `errorOrMigrationWarning` instead of `deprecationWarning`.

OlivierBlanvillain commented:
The CI failure looks unrelated to the diff, I suspect compilation order on the sdtlib:

<details><summary>stacktrace</summary>
<p>
<pre>
exception caught when loading class Repetition: java.lang.AssertionError: assertion failed: scala.collection.parallel.immutable.Repetition.ParIterator$ already has a symbol
assertion failure for scala.collection.parallel.IterableSplitter[T](SplitAt.this.pit) <:< ?{ splitAt2combiners: ? }, frozen = false
[error] Test dotty.tools.dottydoc.TestWhitelistedCollections.arrayAndImmutableHasDocumentation failed: java.lang.AssertionError: assertion failed: scala.collection.parallel.immutable.Repetition.ParIterator$ already has a symbol, took 14.609 sec
[error]     at scala.Predef$.assert(Predef.scala:223)
[error]     at dotty.tools.dotc.core.SymbolLoaders$.enterNew(SymbolLoaders.scala:36)
[error]     at dotty.tools.dotc.core.SymbolLoaders$.enterModule(SymbolLoaders.scala:60)
[error]     at dotty.tools.dotc.core.SymbolLoaders$.enterClassAndModule(SymbolLoaders.scala:99)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.enterClassAndModule$1(ClassfileParser.scala:706)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.$anonfun$enterOwnInnerClasses$1(ClassfileParser.scala:720)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.$anonfun$enterOwnInnerClasses$1$adapted(ClassfileParser.scala:714)
[error]     at scala.collection.mutable.HashMap$$anon$2.$anonfun$foreach$3(HashMap.scala:158)
[error]     at scala.collection.mutable.HashTable.foreachEntry(HashTable.scala:237)
[error]     at scala.collection.mutable.HashTable.foreachEntry$(HashTable.scala:230)
[error]     at scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:44)
[error]     at scala.collection.mutable.HashMap$$anon$2.foreach(HashMap.scala:158)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.enterOwnInnerClasses(ClassfileParser.scala:714)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.parseClass(ClassfileParser.scala:156)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.run(ClassfileParser.scala:87)
[error]     at dotty.tools.dotc.core.ClassfileLoader.load(SymbolLoaders.scala:390)
[error]     at dotty.tools.dotc.core.ClassfileLoader.doComplete(SymbolLoaders.scala:385)
[error]     at dotty.tools.dotc.core.SymbolLoader.complete(SymbolLoaders.scala:332)
[error]     at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:236)
[error]     at dotty.tools.dotc.core.SymDenotations$SymDenotation.unforcedDecls(SymDenotations.scala:358)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.typeParamsFromDecls(SymDenotations.scala:1449)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.typeParams(SymDenotations.scala:1462)
[error]     at dotty.tools.dotc.core.TypeApplications$.$anonfun$typeParams$extension$1(TypeApplications.scala:171)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.TypeApplications$.typeParams$extension(TypeApplications.scala:162)
[error]     at dotty.tools.dotc.core.TypeApplications$.$anonfun$appliedTo$extension0$1(TypeApplications.scala:352)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.TypeApplications$.appliedTo$extension0(TypeApplications.scala:351)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.processClassType$1(ClassfileParser.scala:351)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.sig2type$1(ClassfileParser.scala:359)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.sig2type$1(ClassfileParser.scala:390)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.dotty$tools$dotc$core$classfile$ClassfileParser$$sigToType(ClassfileParser.scala:448)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.parseAttribute$1(ClassfileParser.scala:555)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.$anonfun$parseAttributes$4(ClassfileParser.scala:626)
[error]     at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser.parseAttributes(ClassfileParser.scala:625)
[error]     at dotty.tools.dotc.core.classfile.ClassfileParser$$anon$2.complete(ClassfileParser.scala:268)
[error]     at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:236)
[error]     at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:180)
[error]     at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:182)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.signature(Denotations.scala:725)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.matches(Denotations.scala:1089)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.filterDisjoint(Denotations.scala:1106)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.mapInherited(Denotations.scala:1101)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.mapInherited(Denotations.scala:718)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.collect$1(SymDenotations.scala:1690)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.$anonfun$computeNPMembersNamed$1(SymDenotations.scala:1698)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeNPMembersNamed(SymDenotations.scala:1677)
[error]     at dotty.tools.dotc.core.SymDenotations$PackageClassDenotation.recur$3(SymDenotations.scala:2000)
[error]     at dotty.tools.dotc.core.SymDenotations$PackageClassDenotation.computeNPMembersNamed(SymDenotations.scala:2011)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.nonPrivateMembersNamed(SymDenotations.scala:1667)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.membersNamed(SymDenotations.scala:1654)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.findMember(SymDenotations.scala:1702)
[error]     at dotty.tools.dotc.core.Types$Type.go$1(Types.scala:562)
[error]     at dotty.tools.dotc.core.Types$Type.findMember(Types.scala:712)
[error]     at dotty.tools.dotc.core.Types$Type.memberBasedOnFlags(Types.scala:545)
[error]     at dotty.tools.dotc.core.Types$Type.$anonfun$member$1(Types.scala:530)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.Types$Type.member(Types.scala:530)
[error]     at dotty.tools.dotc.typer.ImportInfo.$anonfun$importedImplicits$2(ImportInfo.scala:110)
[error]     at scala.collection.immutable.List.flatMap(List.scala:338)
[error]     at dotty.tools.dotc.typer.ImportInfo.importedImplicits(ImportInfo.scala:109)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:223)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:230)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:230)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:230)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:230)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:230)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:230)
[error]     at dotty.tools.dotc.core.Contexts$Context.implicits(Contexts.scala:230)
[error]     at dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1268)
[error]     at dotty.tools.dotc.typer.Implicits.$anonfun$inferImplicit$4(Implicits.scala:971)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Implicits.$anonfun$inferImplicit$1(Implicits.scala:968)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:964)
[error]     at dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:964)
[error]     at dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:87)
[error]     at dotty.tools.dotc.typer.Implicits.$anonfun$inferView$1(Implicits.scala:623)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Implicits.inferView(Implicits.scala:609)
[error]     at dotty.tools.dotc.typer.Implicits.inferView$(Implicits.scala:608)
[error]     at dotty.tools.dotc.typer.Typer.inferView(Typer.scala:87)
[error]     at dotty.tools.dotc.typer.Typer.adaptToSubType$1(Typer.scala:2838)
[error]     at dotty.tools.dotc.typer.Typer.adaptNoArgsOther$1(Typer.scala:2662)
[error]     at dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:2718)
[error]     at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:2927)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$adapt$3(Typer.scala:2344)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$adapt$1(Typer.scala:2343)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:2343)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2184)
[error]     at dotty.tools.dotc.typer.Typer.typeSelectOnTerm$1(Typer.scala:450)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedSelect$1(Typer.scala:485)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedSelect(Typer.scala:447)
[error]     at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:1992)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2065)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2184)
[error]     at dotty.tools.dotc.typer.Applications.$anonfun$typedApply$1(Applications.scala:808)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:805)
[error]     at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:911)
[error]     at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:803)
[error]     at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:87)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2012)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.reassignmentToVal$1(Typer.scala:659)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedAssign$1(Typer.scala:689)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedAssign(Typer.scala:642)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2018)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Applications.$anonfun$typedApply$13(Applications.scala:895)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Applications.typedOpAssign$1(Applications.scala:888)
[error]     at dotty.tools.dotc.typer.Applications.$anonfun$typedApply$16(Applications.scala:904)
[error]     at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2192)
[error]     at dotty.tools.dotc.typer.Applications.$anonfun$typedApply$15(Applications.scala:905)
[error]     at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2195)
[error]     at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:902)
[error]     at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:803)
[error]     at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:87)
[error]     at dotty.tools.dotc.typer.Typer.typedInfixOp(Typer.scala:1885)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2048)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2151)
[error]     at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2173)
[error]     at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:701)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedBlock$1(Typer.scala:704)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:703)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2019)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedWhileDo$1(Typer.scala:1196)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedWhileDo(Typer.scala:1192)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2026)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2151)
[error]     at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2173)
[error]     at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:701)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedBlock$1(Typer.scala:704)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:703)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2019)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Namer.$anonfun$typedAheadExpr$1(Namer.scala:1094)
[error]     at dotty.tools.dotc.typer.Namer.typedAheadImpl(Namer.scala:1084)
[error]     at dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1094)
[error]     at dotty.tools.dotc.typer.Namer.rhsType$1(Namer.scala:1217)
[error]     at dotty.tools.dotc.typer.Namer.cookedRhsType$1(Namer.scala:1229)
[error]     at dotty.tools.dotc.typer.Namer.lhsType$1(Namer.scala:1230)
[error]     at dotty.tools.dotc.typer.Namer.inferredType$1(Namer.scala:1248)
[error]     at dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:1256)
[error]     at dotty.tools.dotc.typer.Namer.defDefSig(Namer.scala:1325)
[error]     at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:770)
[error]     at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:881)
[error]     at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:796)
[error]     at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:236)
[error]     at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:180)
[error]     at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:182)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.signature(Denotations.scala:725)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.matches(Denotations.scala:1089)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.filterDisjoint(Denotations.scala:1106)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.mapInherited(Denotations.scala:1101)
[error]     at dotty.tools.dotc.core.Denotations$SingleDenotation.mapInherited(Denotations.scala:718)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.collect$1(SymDenotations.scala:1690)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.collect$1(SymDenotations.scala:1685)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.$anonfun$computeNPMembersNamed$1(SymDenotations.scala:1698)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeNPMembersNamed(SymDenotations.scala:1677)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.nonPrivateMembersNamed(SymDenotations.scala:1667)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.collect$1(SymDenotations.scala:1689)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.collect$1(SymDenotations.scala:1685)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.$anonfun$computeNPMembersNamed$1(SymDenotations.scala:1698)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeNPMembersNamed(SymDenotations.scala:1677)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.nonPrivateMembersNamed(SymDenotations.scala:1667)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.membersNamed(SymDenotations.scala:1654)
[error]     at dotty.tools.dotc.core.SymDenotations$ClassDenotation.findMember(SymDenotations.scala:1702)
[error]     at dotty.tools.dotc.core.Types$Type.go$1(Types.scala:562)
[error]     at dotty.tools.dotc.core.Types$Type.findMember(Types.scala:712)
[error]     at dotty.tools.dotc.core.Types$Type.memberBasedOnFlags(Types.scala:545)
[error]     at dotty.tools.dotc.core.Types$Type.$anonfun$member$1(Types.scala:530)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.Types$Type.member(Types.scala:530)
[error]     at dotty.tools.dotc.typer.ProtoTypes$SelectionProto.isMatchedBy(ProtoTypes.scala:154)
[error]     at dotty.tools.dotc.core.TypeComparer.isMatchedByProto(TypeComparer.scala:1311)
[error]     at dotty.tools.dotc.core.TypeComparer.firstTry$1(TypeComparer.scala:274)
[error]     at dotty.tools.dotc.core.TypeComparer.$anonfun$recur$2(TypeComparer.scala:998)
[error]     at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:44)
[error]     at dotty.tools.dotc.core.TypeComparer.recur(TypeComparer.scala:179)
[error]     at dotty.tools.dotc.core.TypeComparer.isSubType(TypeComparer.scala:152)
[error]     at dotty.tools.dotc.core.TypeComparer.isSubType(TypeComparer.scala:162)
[error]     at dotty.tools.dotc.core.TypeComparer.topLevelSubType(TypeComparer.scala:126)
[error]     at dotty.tools.dotc.core.Types$Type.$anonfun$$less$colon$less$1(Types.scala:839)
[error]     at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.core.Types$Type.$less$colon$less(Types.scala:839)
[error]     at dotty.tools.dotc.typer.Typer.adaptNoArgsOther$1(Typer.scala:2645)
[error]     at dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:2718)
[error]     at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:2927)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$adapt$3(Typer.scala:2344)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$adapt$1(Typer.scala:2343)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:2343)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2184)
[error]     at dotty.tools.dotc.typer.Typer.typeSelectOnTerm$1(Typer.scala:450)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedSelect$1(Typer.scala:485)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedSelect(Typer.scala:447)
[error]     at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:1992)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2065)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2184)
[error]     at dotty.tools.dotc.typer.Applications.$anonfun$typedApply$1(Applications.scala:808)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:805)
[error]     at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:911)
[error]     at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:803)
[error]     at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:87)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2012)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedAssign$1(Typer.scala:676)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedAssign(Typer.scala:642)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2018)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2184)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedDefDef$1(Typer.scala:1554)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:1527)
[error]     at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:1999)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2065)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2131)
[error]     at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2173)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedClassDef$1(Typer.scala:1677)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:1583)
[error]     at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2002)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2065)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2131)
[error]     at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2173)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedClassDef$1(Typer.scala:1677)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:1583)
[error]     at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2002)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2065)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2131)
[error]     at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2173)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedPackageDef$1(Typer.scala:1798)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:1791)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2042)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2151)
[error]     at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2173)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typedPackageDef$1(Typer.scala:1798)
[error]     at dotty.tools.dotc.util.Stats$.track(Stats.scala:37)
[error]     at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:1791)
[error]     at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2042)
[error]     at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2066)
[error]     at dotty.tools.dotc.typer.Typer.$anonfun$typed$2(Typer.scala:2100)
[error]     at dotty.tools.dotc.reporting.trace$.apply(trace.scala:40)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2093)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2112)
[error]     at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2184)
[error]     at dotty.tools.dotc.typer.FrontEnd.$anonfun$typeCheck$1(FrontEnd.scala:61)
[error]     at dotty.tools.dotc.typer.FrontEnd.monitor(FrontEnd.scala:35)
[error]     at dotty.tools.dotc.typer.FrontEnd.typeCheck(FrontEnd.scala:59)
[error]     at dotty.tools.dotc.typer.FrontEnd.$anonfun$runOn$6(FrontEnd.scala:89)
[error]     at dotty.tools.dotc.typer.FrontEnd.$anonfun$runOn$6$adapted(FrontEnd.scala:89)
[error]     at scala.collection.immutable.List.foreach(List.scala:392)
[error]     at dotty.tools.dotc.typer.FrontEnd.runOn(FrontEnd.scala:89)
[error]     at dotty.tools.dotc.Run.$anonfun$compileUnits$3(Run.scala:158)
[error]     at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error]     at dotty.tools.dotc.util.Stats$.trackTime(Stats.scala:49)
[error]     at dotty.tools.dotc.Run.$anonfun$compileUnits$2(Run.scala:155)
[error]     at dotty.tools.dotc.Run.$anonfun$compileUnits$2$adapted(Run.scala:153)
[error]     at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
[error]     at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
[error]     at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
[error]     at dotty.tools.dotc.Run.runPhases$1(Run.scala:153)
[error]     at dotty.tools.dotc.Run.$anonfun$compileUnits$1(Run.scala:178)
[error]     at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error]     at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:102)
[error]     at dotty.tools.dotc.Run.compileUnits(Run.scala:133)
[error]     at dotty.tools.dotc.Run.compileSources(Run.scala:120)
[error]     at dotty.tools.dotc.Run.compile(Run.scala:104)
[error]     at dotty.tools.dottydoc.DottyDocTest.checkFiles(DottyDocTest.scala:98)
[error]     at dotty.tools.dottydoc.DottyDocTest.checkFiles$(DottyDocTest.scala:95)
[error]     at dotty.tools.dottydoc.TestWhitelistedCollections.checkFiles(WhitelistedStdLib.scala:7)
[error]     at dotty.tools.dottydoc.TestWhitelistedCollections.arrayAndImmutableHasDocumentation(WhitelistedStdLib.scala:10)
[error]     ...

smarter commented:
> I suspect compilation order on the sdtlib...

The stdlib is compiled in a fixed order since https://github.com/lampepfl/dotty/pull/5604

OlivierBlanvillain commented:
The issue was caused by the call to `ctx.scala2Mode` which lead to a staled symbol. I changed it to only check for the `-language:Scala2` flag and removed `import language.Scala2` entirely.

odersky commented on compiler/test/dotty/tools/dotc/CompilationTests.scala:
```diff
@@ -31,10 +31,10 @@ class CompilationTests extends ParallelTesting {
+  @Test  // enable to test compileStdLib separately with detailed stats
   def compileStdLibOnly: Unit = {
     implicit val testGroup: TestGroup = TestGroup("compileStdLibOnly")
-    compileList("compileStdLib", TestSources.stdLibSources, scala2Mode.and("-migration", "-Yno-inline", "-Ydetailed-stats"))
```
stdlib is already tested with the same options in `pos`. We should only run the test once, since it is expensive. I don't care whether we keep it as part of `pos` or make it a separate test.
